src/usr/local/buildpack/utils/cache.sh

```diff
@@ -85,3 +107,82 @@ function get_cache_fill_level () {
   check BUILDPACK_CACHE_DIR true
   df --output=pcent "${BUILDPACK_CACHE_DIR}" | tr -dc '0-9'
 }
+
+# Will calculate the checksum for the given string
+function calculate_checksum () {
+  local text=$1
+  echo "${text}" | sha1sum | awk '{print $1}'
+}
+
+# Will cache the given folder under a given key in the cache
+# Any existing entry under the given key will be overriden
+# If the key is not passed, then the path will be used as key
+#
+# Only enabled when BUILDPACK_CACHE_DIR is set, noop otherwise
+#
+# Will return the path to the cached folder as tar file
+function cache_folder () {
+  local path=$1
+  local key=${2:-${1}}
+
+  check path true
+  check key true
+
+  if [ -z "${BUILDPACK_CACHE_DIR}" ]; then
+    # BUILD_CACHE_DIR is not set
+    return
+  fi
+
+  local checksum
+  checksum=$(calculate_checksum "${key}")
+
+  local filename="${checksum}/folder.tar"
+  local cache_path="${BUILDPACK_CACHE_DIR}/${filename}"
+
+  # create folder
+  mkdir -p "${BUILDPACK_CACHE_DIR}/${checksum}"
```

Comment:
we need to set writable flag for root group

Reply:
Will use the utils class here.